### PR TITLE
Update Go updates template with the latest process changes

### PR DESCRIPTION
**Pull request by xmudrii**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

I identified some spots that could be optimized while doing the latest Go updates:

- Build and promote `distroless-iptables` images before updating `kubernetes/kubernetes`
  - This reduces amount of `kubernetes/kubernetes` PRs from two to one. This enables us to ship Go updates faster because we don't need to wait for approvals and CI two times
- Add a new step to update `kubekins` and `krte` images after updating the `master` branch in `kubernetes/kubernetes`
  - This step might be required for some tests to pick up the new Go version. For the sake of the CI signal, it makes sense to do it as early as possible, instead of waiting for all cherry-picks to be done. After merging cherry-picks, we update `kubekins` and `krte` images for release branches
- The list is now condensed. Items that were updated with a single PR are now a single item in the checklist

#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/release/issues/3383

#### Special notes for your reviewer:

I'm working on a new version that's based on task lists, but until then we can update the existing checklist.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/assign @jeremyrickard @cpanato @Verolop @puerco @justaugustus @saschagrunert 
cc @kubernetes/release-engineering 
/hold
for discussion and consensus

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
/lgtm
thanks for updating this

/hold for more 👀

**Review by saschagrunert — APPROVED** [review]
Thanks!

**Comment by xmudrii** [comment]
We have enough approvals, let's get this merged! :shipit: 
/hold cancel

**Pull request merged**
